portfolio/src/main/webapp/script.js

```diff
@@ -32,3 +32,21 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+function fetchTest() {
+  fetch('/data').then(response => response.json()).then((messages) => {
+    console.log(messages);
+    const commentListElement = document.getElementById('helloContainer');
+    commentListElement.innerHTML = '';
+    messages.forEach(element => {
+      commentListElement.appendChild(
+        createListElement('Comment: ' + element.message));
+    });
+  });
+}
+
+function createListElement(text) {
+  const liElement = document.createElement('li');
+  liElement.innerText = text;
```

Comment:
Nice use of createElement() and innerText! Don't take this as a thorough security review but I think this approach is more hardened to [XSS vulnerabilities](https://en.wikipedia.org/wiki/Cross-site_scripting) than one that uses innerHTML directly to set the content

Reply:
I can't take credit for this use of the two since the walkthrough showed us how to create the list elements. I didn't make any changes to the function, but I'll look into what you said and consider it for future use.